Assets/BossRoom/Scripts/Client/UI/Lobby/JoinLobbyUI.cs

```diff
@@ -0,0 +1,172 @@
+using System.Collections.Generic;
+using BossRoom.Scripts.Shared.Infrastructure;
+using BossRoom.Scripts.Shared.Net.UnityServices.Lobbies;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BossRoom.Scripts.Client.UI
+{
+    /// <summary>
+    /// Handles the list of LobbyPanelUIs and ensures it stays synchronized with the lobby list from the service.
+    /// </summary>
+    public class JoinLobbyUI : ObserverBehaviour<LobbyServiceData>
+    {
+        [SerializeField] private LobbyPanelUI m_LobbyPanelPrototype;
+        [SerializeField] private InputField m_JoinCodeField;
+        [SerializeField] private CanvasGroup m_CanvasGroup;
+
+        private IInstanceResolver m_Container;
+
+        /// <summary>
+        /// Key: Lobby ID, Value Lobby UI
+        /// </summary>
+        private readonly Dictionary<string, LobbyPanelUI> m_LobbyButtons = new Dictionary<string, LobbyPanelUI>();
+
+        private LobbyServiceData m_LobbyServiceData;
+        private LobbyUIMediator m_LobbyUIMediator;
+
+        private readonly Dictionary<string, LocalLobby> m_LocalLobby = new Dictionary<string, LocalLobby>();
+
+        private UpdateRunner m_UpdateRunner;
+
+        private void Awake()
+        {
+            m_LobbyPanelPrototype.gameObject.SetActive(false);
+        }
+
+        private void OnDisable()
+        {
+            if (m_UpdateRunner != null)
+            {
+                m_UpdateRunner.Unsubscribe(PeriodicRefresh);
+            }
+        }
+
+        [Inject]
+        private void InjectDependencies(IInstanceResolver container, LobbyUIMediator lobbyUIMediator, LobbyServiceData lobbyServiceData, UpdateRunner updateRunner)
```

Comment:
```suggestion
        private void InjectDependenciesAndInitialize(IInstanceResolver container, LobbyUIMediator lobbyUIMediator, LobbyServiceData lobbyServiceData, UpdateRunner updateRunner)
```
Adding "initialize" since this also subscribes to update runner